src/hmm/transition-model.h

```diff
@@ -315,7 +323,7 @@ inline int32 TransitionModel::TransitionIdToPdf(int32 trans_id) const {
   KALDI_ASSERT(static_cast<size_t>(trans_id) < id2state_.size() &&
                "Likely graph/model mismatch (graph built from wrong model?)");
   int32 trans_state = id2state_[trans_id];
-  return triples_[trans_state-1].pdf;
```

Comment:
Regarding TransitionIdToPdf(), the speed is going to be a problem if there is going to be an extra if-statement in there, as this gets called very frequently during decoding.  It will be better if you introduce a new derived parameter called id2pdf_, mapping from transition-id directly to pdf, then it's just a lookup.